### PR TITLE
[Chore](build) enable -Wpedantic and update lowest gcc version to 11.1

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

1. enable `-Wpedantic` on clang
2. remove some extra `;`
3. update lowest gcc version from `11.0` to `11.1` and remove compatibility code
4. ~fix `StringRef::MAX_CHAR` overflow~
5. modify clangd config to make `-fno-access-control` work on unittest.
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
5. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.95 seconds
 load time: 498 seconds
 storage size: 17122663528 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230201122046_clickbench_pr_88878.html

File: be/src/olap/rowset/unique_rowset_id_generator.h [hunk 1]
```diff
@@ -28,7 +28,10 @@ namespace doris {
 class UniqueRowsetIdGenerator : public RowsetIdGenerator {
 public:
     UniqueRowsetIdGenerator(const UniqueId& backend_uid);
-    ~UniqueRowsetIdGenerator();
+    ~UniqueRowsetIdGenerator() override;
+
+    UniqueRowsetIdGenerator(const UniqueRowsetIdGenerator&) = delete;
+    UniqueRowsetIdGenerator operator=(const UniqueRowsetIdGenerator&) = delete;
```
Comment (adonis0147) [inline be/src/olap/rowset/unique_rowset_id_generator.h:34]:
```suggestion
    UniqueRowsetIdGenerator& operator=(const UniqueRowsetIdGenerator&) = delete;
```
See [https://en.cppreference.com/w/cpp/language/copy_assignment](https://en.cppreference.com/w/cpp/language/copy_assignment)

**Review by adonis0147 — DISMISSED** [review]
LGTM

File: be/src/vec/common/string_ref.h [hunk 2]
```diff
@@ -242,8 +243,8 @@ struct StringRef {
     StringRef trim() const;
 
     // support for type_limit
-    static constexpr char MIN_CHAR = 0x00;
-    static constexpr char MAX_CHAR = 0xFF;
+    static constexpr char MIN_CHAR = CHAR_MIN;
+    static constexpr char MAX_CHAR = CHAR_MAX;
```
Comment (HappenLee) [inline be/src/vec/common/string_ref.h:247]: Should be 0xFF， not `CHAR_MAX`

**Review by HappenLee — CHANGES_REQUESTED** [review]
Danger to change `MAX_CHAR`

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**